amethyst_controls/src/systems.rs

```diff
@@ -80,6 +80,32 @@ where
     }
 }
 
+/// The system that manages the arc ball movement;
+pub struct ArcBallMovementSystem;
+
+impl<'a> System<'a> for ArcBallMovementSystem {
+    type SystemData = (
+        WriteStorage<'a, Transform>,
+        ReadStorage<'a, ArcBallCameraTag>,
+    );
+
+    fn run(&mut self, (mut transforms, tags): Self::SystemData) {
+        let mut position = None;
+        for (transform, arc_ball_camera_tag) in (&transforms, &tags).join() {
+            if let Some(target_transform) = transforms.get(arc_ball_camera_tag.target) {
+                let target_to_cam = transform.translation - target_transform.translation;
+                let new_target_to_cam = target_to_cam.normalize() * arc_ball_camera_tag.distance;
+                position = Some(target_transform.translation + new_target_to_cam);
+            }
+        }
+        if let Some(new_pos) = position {
```

Comment:
This won't work for multiple arc ball cameras, I wonder if we should support that.

Reply:
You should only be controlling a single camera at once normally. If you want multi controller support input events will need to have a tracker of which controller/mouse does what.